src/Http/Http.Abstractions/src/Extensions/HttpResponseWritingExtensions.cs

```diff
@@ -66,6 +66,11 @@ public static Task WriteAsync(this HttpResponse response, string text, Encoding
                 throw new ArgumentNullException(nameof(encoding));
             }
 
+            if(response.ContentType == null)
```

Comment:
WriteAsync can be called multiple times, so this would fail the second time. This should move to the caller of WriteAsync.